torch/_library/opaque_object.py

```diff
@@ -3,14 +3,97 @@
 import torch
 
 
-def make_opaque(payload: Any) -> torch._C.ScriptObject:
+def make_opaque(payload: Any = None) -> torch._C.ScriptObject:
     """
     Creates an opaque object which stores the given Python object.
     This opaque object can be passed to any custom operator as an argument.
     The Python object can then be accessed from the opaque object using the `get_payload()` API.
+    The opaque object has `._type()`
+    "__torch__.torch.classes.aten.OpaqueObject", which should be the type used
+    when creating custom operator schemas.
+
+    Args:
+        payload (Any): The Python object to store in the opaque object. This can
+        be empty, and can be set with `set_payload()` later.
+
+    Returns:
+        torch._C.ScriptObject: The opaque object that stores the given Python object.
+
+    Example:
+
+        >>> import torch
+        >>> from torch._library.opaque_object import (
+        ...     make_opaque,
+        ...     get_payload,
+        ...     set_payload,
+        ... )
+        >>>
+        >>> class OpaqueQueue:
+        >>>     def __init__(self, queue: list[torch.Tensor], init_tensor_: torch.Tensor) -> None:
+        >>>         super().__init__()
+        >>>         self.queue = queue
+        >>>         self.init_tensor_ = init_tensor_
```

Comment:
I would not recommend OpaqueQueue as the example for users. Might be better to include something that isn't a container of tensors.

For example, maybe Point with x, y being floats.